source/game/kart/KartState.cc

```diff
@@ -339,6 +339,10 @@ void KartState::calcCollisions() {
             m_bGroundStart = true;
         }
 
+        if (!state()->isJumpPad()) {
+            m_bJumpPadMushroomCollision = false;
+        }
+
```

Comment:
```suggestion
        if (!m_bJumpPad) {
            m_bJumpPadMushroomCollision = false;
        }

```
This should be added before the `if (!wasTouchingGround)` check.